src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -4958,6 +4963,9 @@ void CalcZoneLeavingConditions(EnergyPlusData &state, bool const FirstHVACIterat
                     } else {
                         state.dataLoopNodes->Node(ReturnNode).Temp = TempRetAir;
                     }
+                    if (ReturnNodeExhaustNum > 0 && state.dataLoopNodes->Node(ReturnNodeExhaustNum).MassFlowRate > 0.0 && QRetAir > 0.0) {
+                        state.dataLoopNodes->Node(ReturnNodeExhaustNum).Temp = TempRetAir;
+                    }
```

Comment:
Assign node temperature after adding return heat to both nodes of return and zone exhaust.